x-pack/solutions/security/plugins/session_view/public/components/tty_player/hooks.ts

```diff
@@ -318,21 +318,26 @@ export const useXtermPlayer = ({
 
   useEffect(() => {
     if (isPlaying) {
-      const timer = setTimeout(() => {
-        if (!hasNextPage && currentLine === lines.length - 1) {
-          setIsPlaying(false);
-        } else {
-          const nextLine = Math.min(lines.length - 1, currentLine + 1);
-          render(nextLine, false);
-          setCurrentLine(nextLine);
-        }
-      }, playSpeed);
+      const timer = setInterval(
+        () =>
+          setCurrentLine((_currentLine) => {
+            if (!hasNextPage && _currentLine === lines.length - 1) {
+              setIsPlaying(false);
+              return _currentLine;
+            } else {
+              const nextLine = Math.min(lines.length - 1, _currentLine + 1);
+              render(nextLine, false);
+              return nextLine;
+            }
+          }),
+        playSpeed
+      );
 
       return () => {
-        clearTimeout(timer);
+        clearInterval(timer);
       };
     }
-  }, [lines, currentLine, isPlaying, playSpeed, render, hasNextPage, fetchNextPage, setIsPlaying]);
+  }, [lines, isPlaying, playSpeed, render, hasNextPage, fetchNextPage, setIsPlaying]);
```

Comment:
opting for this approach means we aren't continuously setting up the timeout and destroying it when the current line changes especially because  it gets set in this useEffect and is also a dependency of this effect, with the proposed approach we set up an interval that runs in perpetuity as long as the terminal is playing